reference/extensions/deployers.rst

```diff
@@ -1,5 +1,76 @@
 .. _reference_extensions_deployers:
 
 Deployers
----------
+=========
 
+Deployers are a mechanism to facilitate copying files form one folder, usually the Conan cache, to user folders.
+While Conan provides two built-in ones (``full_deploy`` and ``direct_deploy``), users can easily manage their own
+with ``conan config install``.
+
+Deployers run before generators, and they can change the target folders.
+For example, if the ``--deploy=full_deploy`` deployer runs before ``CMakeDeps``,
+the files generated by ``CMakeDeps`` will point to the local copy in the user folder done by the ``full_deploy`` deployer,
+and not to the Conan cache. Multiple deployers can be specified by supplying more than one ``--deploy=`` argument,
+and they will be ran in order of appearance.
+
+Deployers can be multi-configuration. Running ``conan install . --deploy=full_deploy`` repeatedly for different profiles
+can achieve a fully self-contained project, including all the artifacts, binaries, and build files.
+This project would be completely independent of Conan and no longer require it at all to build.
+
+
+Built-in deployers
+------------------
+
+.. _reference_extensions_deployer_full_deploy:
+
+full_deploy()
+^^^^^^^^^^^^^
+
+Deploys each package folder of every dependency to your recipe's ``output_folder`` in a subfolder tree based on:
+
+#. The build context
+#. The dependency name and version
+#. The build type
+#. The build arch
+
+Such that every dependency will end up in a folder such as:
+
+``[OUTPUT_FOLDER]/host/dep/0.1/Release/x86_64``
```

Comment:
as it doesn't include all the settings/options, is there a risk that several deployments will overwrite each other? maybe it's better just to use package id here?

Reply:
There is, but if the user is worried about such problems, they are encouraged to write their own deployer, but let's tag @memsharded in case he thinks this could be nice to have